### PR TITLE
Add Doc for Local Development with Managed Authentication

### PR DESCRIPTION
**Pull request by ThumulaPerera**
## Purpose
$subject

Resolves: https://github.com/wso2-enterprise/choreo/issues/26188

## Related PRs
https://github.com/wso2-enterprise/choreo-console/pull/8836

### CODE REVIEW

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+However, as a web application developer, you would also need to run the application locally on your personal workstation. You’d rightly expect the authentication and authorization to work when developing locally just as it would when the application is deployed to Choreo, without having to change the source code of your application. (If you have previously used any OIDC/OAuth2 SDKs to add authentication/authorization to a web app, you know that logins, APIs calls work when running the application locally. But there you’d have to configure the Client Secrets locally). 
+
+Local development provides the same capabilities when developing the web application locally.
+
+## How local development works
+
+Local development uses a proxy that sits infront of the locally running web application. This proxy is responsible for forwarding the requests to managed authentiication BFF endpoints (`/choreo-apis/*`, `/auth/*`) to Choreo and all other requests to the locally running web application. The proxy runs on https (using a self signed certificate) as managed authentication relies on secure, http-only cookies.
```
Comment (Nashaath) [inline en/docs/authentication-and-authorization/local-development-with-managed-authentication.md:11]: Should the second sentence be "This proxy is responsible for forwarding the requests from the managed authentiication BFF endpoints (/choreo-apis/*, /auth/*) to Choreo and all other requests to the locally running web application."

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Local development uses a proxy that sits infront of the locally running web application. This proxy is responsible for forwarding the requests to managed authentiication BFF endpoints (`/choreo-apis/*`, `/auth/*`) to Choreo and all other requests to the locally running web application. The proxy runs on https (using a self signed certificate) as managed authentication relies on secure, http-only cookies. 
+After running the proxy, the developer can access the web app using the proxy's URL and continue development and the behaviour of the web app (in terms of managed authentication functionality) would be similar to how it would be after deplying to Choreo.  
+
+
+## Configure Local Development
+
+### Prerequisites
+
+- A SPA web application with managed authentication enabled.
```
Comment (senthalan) [inline en/docs/authentication-and-authorization/local-development-with-managed-authentication.md:19]: We used two different ways `managed authentication` and `Managed Authentication`  better to be consistent. I think we can use `Managed Authentication`

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+### Develop your web app locally
+
+=== "Using Choreo built-in identity provider"
+
+    1. Click local development on the environment card to open the right drawer.
+    2. Copy the command given in step 1 on the right drawer. 
+    3. Relace [APP_PORT] with the port on which your application is running locally. 
+    4. Run the command in a terminal.
+    5. Access the application using the URL given in step 2 on the right drawer and start/continue development.
```
Comment (senthalan) [inline en/docs/authentication-and-authorization/local-development-with-managed-authentication.md:63]: ` and start/continue development.` shall we reword it ?

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md (hunk 1 above)
Comment (senthalan) [reply en/docs/authentication-and-authorization/local-development-with-managed-authentication.md:11]: `authentiication` -> `authentication`

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# Develop web applications locally with Choreo’s Managed Authentication
+
+The Managed Authentication capability of Choreo exposes a set of BFF endpoints which can be inoked by web applications to perform authentication and authorization. These endpoints are readily available for single page web applications which are deployed on Choreo. 
+
+However, as a web application developer, you would also need to run the application locally on your personal workstation. You’d rightly expect the authentication and authorization to work when developing locally just as it would when the application is deployed to Choreo, without having to change the source code of your application. (If you have previously used any OIDC/OAuth2 SDKs to add authentication/authorization to a web app, you know that logins, APIs calls work when running the application locally. But there you’d have to configure the Client Secrets locally).
```
Comment (senthalan) [inline en/docs/authentication-and-authorization/local-development-with-managed-authentication.md:5]: `But there you’d have to configure the Client Secrets locally` Is this statement correct?

Comment (ThumulaPerera) [reply en/docs/authentication-and-authorization/local-development-with-managed-authentication.md]: Removed this unnecessary comparison with [Fix comments](https://github.com/wso2/docs-choreo-dev/pull/1361/commits/f15a4e95769b2929db826855cd995a5504f1e3e5)

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md (hunk 2 above)
Comment (ThumulaPerera) [reply en/docs/authentication-and-authorization/local-development-with-managed-authentication.md]: Resolved with [Fix comments](https://github.com/wso2/docs-choreo-dev/pull/1361/commits/f15a4e95769b2929db826855cd995a5504f1e3e5)

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
+# Develop Web Applications Locally with Choreo’s Managed Authentication
+
+Choreo’s  Managed Authentication capability exposes a set of backend for frontend (BFF) endpoints, facilitating authentication and authorization for web applications. These endpoints are readily available for single-page web applications deployed on Choreo.
+
+As a web application developer, there may be instances where it becomes necessary to run the application on your personal workstation. In such situations, the expectation is for authentication and authorization during local development to work in the same manner as it would when the application is deployed to Choreo, eliminating the need to modify the source code (If you have used OIDC/OAuth2.0 SDKs to integrate authentication and authorization into a web application, you are likely familiar with the common requirement to configure Client Secrets locally for enabling logins and API calls when running the application locally).
```
Comment (ThumulaPerera) [inline en/docs/authentication-and-authorization/local-development-with-managed-authentication.md:5]: @Nashaath shall we remove the part within brackets. We decided to remove that comparison as it seems unnecessary here

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+Local development in Choreo provides the same capabilities as developing the web application locally.
+
+## How local development works in Choreo
+
+Local development uses a proxy that sits in front of the locally running web application. This proxy forwards requests to managed authentication BFF endpoints `(/choreo-apis/*, /auth/*)`. Subsequently, these requests are routed to Choreo, while all other incoming requests are forwarded to the locally running web application. The proxy runs on HTTPS using a self-signed certificate, crucial for secure, HTTP-only cookie-based managed authentication. Upon running the proxy, you can access the web application using the proxy's URL and start developing the application locally. You will observe that the behavior of the web application, in terms of the managed authentication functionality, is similar to how it would be after deploying to Choreo.
```
Comment (ThumulaPerera) [inline en/docs/authentication-and-authorization/local-development-with-managed-authentication.md:11]: This proxy forwards requests to managed authentication BFF endpoints `(/choreo-apis/*, /auth/*)`. Subsequently, these requests are routed to Choreo, while all other incoming requests are forwarded to the locally running web application.

->

This proxy intercepts all incoming requests. The requests to `/choreo-apis/*` and /auth/*` are then forwarded to Choreo, while all other requests are forwarded to locally running web application.

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md (hunk 5 above)
Comment (Nashaath) [reply en/docs/authentication-and-authorization/local-development-with-managed-authentication.md]: Done.

File: en/docs/authentication-and-authorization/local-development-with-managed-authentication.md (hunk 6 above)
Comment (Nashaath) [reply en/docs/authentication-and-authorization/local-development-with-managed-authentication.md]: +1
Done.

**Review by Nashaath — APPROVED** [review]

File: en/docs/develop-components/develop-web-applications/develop-web-applications-locally-with-managed-authentication.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
+# Develop Web Applications Locally with Choreo’s Managed Authentication
+
+Choreo’s  Managed Authentication capability exposes a set of backend for frontend (BFF) endpoints, facilitating authentication and authorization for web applications. These endpoints are readily available for single-page web applications deployed on Choreo.
+
+As a web application developer, there may be instances where it becomes necessary to run the application on your personal workstation. In such situations, the expectation is for authentication and authorization during local development to work in the same manner as it would when the application is deployed to Choreo, eliminating the need to modify the source code.
+
+Local development in Choreo provides the same capabilities as developing the web application locally.
```
Comment (ThumulaPerera) [inline en/docs/develop-components/develop-web-applications/develop-web-applications-locally-with-managed-authentication.md:7]: There seems to be a disconnect with the previous paragraph and this. WDYT @Nashaath?

Comment (Nashaath) [reply en/docs/develop-components/develop-web-applications/develop-web-applications-locally-with-managed-authentication.md:7]: Shall we remove this sentence and perhaps have a para describing Choreo's local development capability? Something along the lines of 
"Choreo's local development capability ensures a seamless local application development experience where the authentication and authorization process aligns with Choreo’s deployment environment. This facilitates a smoother transition from development to deployment, improving overall workflow efficiency."

**Review by Nashaath — APPROVED** [review]

**Pull request merged**